src/Nethermind/Nethermind.Taiko/TaikoGasLimitCalculator.cs

```diff
@@ -0,0 +1,15 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Consensus;
+using Nethermind.Core;
+
+namespace Nethermind.Taiko;
+
+/// This class can be refactored together with OptimismGasLimitCalculator into a common GasLimitCalculator
+internal class TaikoGasLimitCalculator : IGasLimitCalculator
+{
+    public long GetGasLimit(BlockHeader parentHeader) =>
+        throw new InvalidOperationException("GasLimit in Taiko should not be derived from parent header.");
```

Comment:
gas limit calculator can return whatever, it doesn't have to derive from parent, not sure what the intention is here.